src/utils/misc.js

```diff
@@ -0,0 +1,6 @@
+/* eslint-disable import/prefer-default-export */
+/**
+ * Module operator.
+ * When `num < 0`, works different than raw remainder operator: num % divider
+ * */
+export const mod = (num, divider) => ((num % divider) + divider) % divider;
```

Comment:
👍 , took a while for me to understand the purpose. It's used for the case go to the past, with left offset.